app/kuma-dp/pkg/dataplane/metrics/server.go

```diff
@@ -243,3 +356,47 @@ func (s *Hijacker) passRequestHeaders(into http.Header, from http.Header) {
 func (s *Hijacker) NeedLeaderElection() bool {
 	return false
 }
+
+const (
+	hdrContentType = "Content-Type"
+	textType       = "text/plain"
+)
+
+// responseFormat extracts the correct format from a HTTP response header.
+// If no matching format can be found FormatUnknown is returned.
+func responseFormat(h http.Header) expfmt.Format {
+	ct := h.Get(hdrContentType)
+
+	mediatype, params, err := mime.ParseMediaType(ct)
+	if err != nil {
+		return expfmt.FmtUnknown
+	}
+
+	version := params["version"]
+
+	switch mediatype {
+	case expfmt.ProtoType:
+		p := params["proto"]
+		e := params["encoding"]
+		// only delimited encoding is supported by prometheus scraper
+		if p == expfmt.ProtoProtocol && e == "delimited" {
+			return expfmt.FmtProtoDelim
+		}
+
+	// if mediatype is `text/plain`, return Prometheus text format
+	// without checking the version, as there are few exporters
+	// which don't set the version param in the content-type header. ex: Envoy
+	case textType:
+		return expfmt.FmtText
+
+	case expfmt.OpenMetricsType:
+		if version == expfmt.OpenMetricsVersion_0_0_1 {
+			return expfmt.FmtOpenMetrics_0_0_1
+		}
+		if version == expfmt.OpenMetricsVersion_1_0_0 {
+			return expfmt.FmtOpenMetrics_1_0_0
+		}
```

Comment:
I think I would go with switch statement in the case of openmetrics and also allow empty version value, following the [prometheus common](https://github.com/prometheus/common/blob/main/expfmt/encode.go#L106)
```suggestion
	switch mediatype {
	case expfmt.ProtoType:
		p := params["proto"]
		e := params["encoding"]
		// only delimited encoding is supported by prometheus scraper
		if p == expfmt.ProtoProtocol && e == "delimited" {
			return expfmt.FmtProtoDelim
		}

	// if mediatype is `text/plain`, return Prometheus text format
	// without checking the version, as there are few exporters
	// which don't set the version param in the content-type header. ex: Envoy
	case textType:
		return expfmt.FmtText

	case expfmt.OpenMetricsType:
		switch params["version"] {
		case expfmt.OpenMetricsVersion_1_0_0:
			return expfmt.FmtOpenMetrics_1_0_0
		case expfmt.OpenMetricsVersion_0_0_1, "":
			return expfmt.FmtOpenMetrics_0_0_1
		}
```

Reply:
I'm gonna approve the PR after resolving this one as I think we should return correct openmetrics format if the value parameter is an empty string, following the prometheus .

Reply:
Okay. I think I asked earlier once, if we don't find version param in Content-Type should we return `FmtUnknown`.
Anyway, I also think we should handle this case.

As per Prometheus, should we just do this instead of switch case even?:
```go
if ver == OpenMetricsVersion_1_0_0 {
	return FmtOpenMetrics_1_0_0
}
return FmtOpenMetrics_0_0_1
```

Reply:
Once this conversation is resolved, will push the requested changes all at once.

Reply:
I think it should do the same as prometheus, so if type is openmetrics, but version is empty, then we should return `FmtOpenMetrics_0_0_1`.
I don't mind the style, but be aware that proposed by you 
```go
if ver == OpenMetricsVersion_1_0_0 {
	return FmtOpenMetrics_1_0_0
}
return FmtOpenMetrics_0_0_1
```
it not working in the same way as it would accept any value of `version`, not only `1.0.0`, `0.0.1` and `""`. I believe if the version is not one of these, we should return `FmtUnknown`.

Reply:
Yeah, right. Let me do the changes.